### ISSUE TITLE
Docker containers not removed by image name alone 

### ISSUE THREAD

**Issue by metajungle** [ansible/ansible-modules-core#2948]
##### ISSUE TYPE

Bug Report
##### COMPONENT NAME

_docker module
##### ANSIBLE VERSION

N/A
##### SUMMARY

The following command does not remove running containers started from a given image, which would be nice to have: 

```
$ ansible -i prod docker-hosts -s -m docker -a "image=repo/image-name:latest state=absent"
10.12.6.31 | SUCCESS => {
    "ansible_facts": {
        "docker_containers": []
    }, 
    "changed": false, 
    "msg": "No action taken.", 
    "reload_reasons": null, 
    "summary": {
        "created": 0, 
        "killed": 0, 
        "pulled": 0, 
        "removed": 0, 
        "restarted": 0, 
        "started": 0, 
        "stopped": 0
    }
}
```

However, if I add the name of the container, it is stopped and removed: 

```
$ ansible -i prod docker-hosts -s -m docker -a "image=repo/image-name:latest name=name_of_container state=absent"
10.12.6.31 | SUCCESS => {
    ...
    "changed": true, 
    "msg": "removed 1 container, stopped 1 container.", 
    "reload_reasons": null, 
    "summary": {
        "created": 0, 
        "killed": 0, 
        "pulled": 0, 
        "removed": 1, 
        "restarted": 0, 
        "started": 0, 
        "stopped": 1
    }
}
```

It would be nice to be able to stop containers based on the images they are launched from, without having to explicitly specify the name.

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29909
